app/models/project.rb

```diff
@@ -40,6 +40,8 @@ class Project < ApplicationRecord
     where("r1.project_id = :id OR r2.project_id = :id", id: project.id)
   }
 
+  has_one :key_set, class_name: 'Dataset::KeySet', as: :resource
```

Comment:
For now just a has one association but can be changed to has many in the future when needed.